### PR TITLE
fix: infinite recursion bug

### LINKED ISSUE #20249: unsupported value: +Inf" error not handled gracefully
JSON marshalling errors should be returned as properly formatted JSON errors.

__Steps to reproduce:__
```
> create database db
> use db
Using database db
> insert m v=1.7976931348623157E308
> insert m v=1.7976931348623157E308
> 
> select * from m
name: m
time                v
----                -
1604428620720870437 1.7976931348623157e+308
1604428621584887557 1.7976931348623157e+308

> select sum(v) from m
ERR: 
> exit
$ curl "http://localhost:8086/query?db=EAR1834" --data-urlencode "q=select sum(v) from m"  
```

__Expected behavior:__
```
$ curl "http://localhost:8086/query?db=EAR1834" --data-urlencode "q=select sum(v) from m"  
{"error":"json: unsupported value: +Inf"}
```

__Actual behavior:__
```
$ curl "http://localhost:8086/query?db=db" --data-urlencode "q=select sum(v) from m"  
json: error calling MarshalJSON for type httpd.Response: json: error calling MarshalJSON for type *query.Result: json: unsupported value: +Inf

```
__Environment info:__

* v1.8.3

### PR DESCRIPTION
**Pull request by lesam**
Closes #20249

We should avoid infinite recursion!

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: services/httpd/response_writer.go [hunk 1]
```diff
@@ -98,13 +98,9 @@ func (w *bytesCountWriter) Write(data []byte) (int, error) {
 
 // WriteResponse writes the response using the formatter.
 func (w *responseWriter) WriteResponse(resp Response) (int, error) {
-	n := 0
 	writer := bytesCountWriter{w: w.ResponseWriter}
 	err := w.formatter.WriteResponse(&writer, resp)
-	if err != nil {
-		n, _ = WriteError(w, err)
```
Comment (lesam) [inline services/httpd/response_writer.go:105]: WriteError calls WriteResponse, and if the ResponseWriter is bad and returning errors, that calls back to WriteError. This was introduced recently, never released.

**Review by davidby-influx — DISMISSED** [review]
LGTM.  Thanks!

File: services/httpd/response_writer.go [hunk 2]
```diff
@@ -131,15 +132,31 @@ func (f *jsonFormatter) WriteResponse(w io.Writer, resp Response) (err error) {
...
+			b, _ = json.MarshalIndent(resp, "", "    ")
+		} else {
+			b, _ = json.Marshal(resp)
+		}
+		w.Write(b)
+		w.Write([]byte("\n"))
+		return err
 	}
 
+	_, err = w.Write(b)
```
Comment (davidby-influx) [inline services/httpd/response_writer.go:148]: Now wondering if we should check for an error on the w.Write ([]byte("\n") call.  Or maybe append the \n and only have one write (in both places it is done).

**Review by davidby-influx — DISMISSED** [review]
One comment to ponder, but not blocking.

File: services/httpd/response_writer.go (hunk 2 above)
Comment (lesam) [reply services/httpd/response_writer.go:149]: Fixed the error check

**Review by davidby-influx — APPROVED** [review]
All the right errors returned, all the wrong errors ignored!

**Pull request merged**